docs/architecture/product_modelling/imports.md

```diff
@@ -0,0 +1,99 @@
+input from a form page, and then interacts with external systems to provision a new node.
+
+For the `ImportedNode`, the creation workflow is `create_imported_node`. This workflow can be designed to still take
+user input from a form page, but could also be provided programmatically by either a CLI command or API endpoint that is
+called. Data sources can include external API resources, CSV- or YAML files, etc. If desired, interaction with all
+external provisioning systems can be skipped, resulting in a creation workflow that could be as simple as follows.
+
+```python
+from orchestrator.workflow import StepList, begin, done, workflow
+from orchestrator.workflows.steps import resync, store_process_subscription
+
+@workflow("Create imported Node", target=Target.CREATE)
+def create_imported_node() -> StepList:
+    """Workflow to import a Node without provisioning it."""
+    return (
+        begin
+        >> create_subscription
+        >> store_process_subscription(Target.CREATE)
+        >> initialize_subscription
+        >> set_status(SubscriptionLifecycle.ACTIVE)
+        >> resync
+        >> done
+    )
+```
+
+## Importing Products
+With the `ImportedNode` part of the service database, we need a modification workflow to take the imported product to a
+`Node` subscription. This is done using a modification workflow `import_node` that is added to the `ImportedNode`
+product.
+
+This workflow is another place where external provisioning could take place, but it could also be a
+straightforward workflow alike the example given earlier. For the modification of importing the `Node` product, the
+following serves as an example.
+
+```python
+from orchestrator.workflow import StepList, begin, done, step, workflow
+from orchestrator.workflows.steps import resync, store_process_subscription, unsync
+
+@step("Create new Node subscription")
+def import_node_subscription(subscription_id: UUIDstr) -> State:
+    """Take an ImportedNode subscription, and turn it into a Node subscription."""
+    imported_node = ImportedNode.from_subscription(subscription_id)
+    new_subscription_id = get_product_id_by_name(ProductName.NODE)
+    new_subscription = Node.from_other_product(imported_node, new_subscription_id)
+
+    return {"subscription": new_subscription}
+
+
+@workflow("Import Node", target=Target.MODIFY)
+def import_node() -> StepList:
+    """Modify into a Node subscription to complete the import."""
+    return (
+        begin
+        >> store_process_subscription(Target.MODIFY)
+        >> unsync
+        >> import_node_subscription
+        >> resync
+        >> done
+    )
+```
+
+In this workflow, the existing `ImportedNode` subscription is modified into a `Node` subscription, and is stored in the
+service database. Now, the import has been completed, and the imported variety of the product has been replaced with a
+"regular" `Node` subscription.
```

Comment:
Is the ImportedNode subscription actually gone after this workflow?

Reply:
Fortunately, yes